_inc/client/components/plans/plan-price/index.jsx

```diff
@@ -14,72 +14,94 @@ import { getCurrencyObject } from '@automattic/format-currency';
+	getPriceRange() {
+		const { currencyCode, rawPrice } = this.props;
 
-		if ( ! currencyCode || ! rawPrice ) {
-			return null;
-		}
-
-		// "Normalize" the input price or price range.
-		const rawPriceRange = Array.isArray( rawPrice ) ? rawPrice.slice( 0, 2 ) : [ rawPrice ];
-		if ( rawPriceRange.includes( 0 ) ) {
-			return null;
+		// "Normalize" the input price or price range
+		let rawPriceRange;
+		if ( Array.isArray( rawPrice ) ) {
+			const positivePrices = rawPrice.filter( price => price >= 0 );
+			// First array entry is lowest price, second is highest price.
+			rawPriceRange = [ Math.min( ...positivePrices ), Math.max( ...positivePrices ) ];
+		} else {
+			rawPriceRange = [ rawPrice ];
 		}
 
-		const priceRange = rawPriceRange.map( item => {
+		return rawPriceRange.map( item => {
 			return {
 				price: getCurrencyObject( item, currencyCode ),
 				raw: item,
 			};
 		} );
+	}
 
-		const classes = classNames( 'plan-price', className, {
-			'is-original': original,
-			'is-discounted': discounted,
-		} );
-
-		const renderPriceHtml = priceObj => {
-			return (
-				<>
-					<span className="plan-price__integer">{ priceObj.price.integer }</span>
-					<sup className="plan-price__fraction">
-						{ priceObj.raw - priceObj.price.integer > 0 && priceObj.price.fraction }
-					</sup>
-				</>
-			);
-		};
+	renderPrice( priceObj ) {
+		return (
+			<>
+				<span className="plan-price__integer">{ priceObj.price.integer }</span>
+				<sup className="plan-price__fraction">
+					{ priceObj.raw - priceObj.price.integer > 0 && priceObj.price.fraction }
+				</sup>
+			</>
+		);
+	}
 
-		const smallerPriceHtml = renderPriceHtml( priceRange[ 0 ] );
-		const higherPriceHtml = priceRange[ 1 ] && renderPriceHtml( priceRange[ 1 ] );
+	renderContent() {
+		const priceRange = this.getPriceRange();
+		const smallerPrice = this.renderPrice( priceRange[ 0 ] );
+		const higherPrice = priceRange[ 1 ] && this.renderPrice( priceRange[ 1 ] );
 
 		return (
-			<h4 className={ classes }>
+			<>
 				<sup className="plan-price__currency-symbol">{ priceRange[ 0 ].price.symbol }</sup>
-				{ ! higherPriceHtml && renderPriceHtml( priceRange[ 0 ] ) }
-				{ higherPriceHtml &&
+				{ ! higherPrice && this.renderPrice( priceRange[ 0 ] ) }
+				{ higherPrice &&
 					__( '{{smallerPrice/}}-{{higherPrice/}}', {
-						components: { smallerPrice: smallerPriceHtml, higherPrice: higherPriceHtml },
+						components: {
+							smallerPrice,
+							higherPrice,
+						},
 						comment: 'The price range for a particular product',
 					} ) }
-			</h4>
+			</>
+		);
+	}
+
+	render() {
+		const { className, discounted, inline, original, rawPrice } = this.props;
+
+		if ( rawPrice === undefined ) {
+			return null;
+		}
+
+		const classes = classNames( 'plan-price', className, {
+			'is-discounted': discounted,
+			'is-inline': inline,
+			'is-original': original,
+		} );
+
+		return inline ? (
+			<span className={ classes }>{ this.renderContent() }</span>
```

Comment:
Seems like to avoid repetition we could render the same thing but keep the component a variable:

For example (untested)

```
const WrapperComponent = inline ? <span> : <div>;
return <WrapperComponent className={ classes }> { this.renderContent() }</WrapperComponent>
```

Reply:
This doesn't seem to work, even after closing the tags (`<span /> : `<div />`). I'm not sure if there's a way to simplify this statement.

Reply:
Yeah, I haven't tested it 😅 But I've done this before, maybe something like:

```
const WrapperComponent = inline ? span : div;
return <WrapperComponent className={ classes }> { this.renderContent() }</WrapperComponent>
```

Reply:
I think that's the last thing I'd like to do before shipping this one.

Reply:
@tyxla I don't really know how this could work. `span` and `div` here would have to be some kind of keywords, otherwise we're getting errors.

The only way I was able to make it work is like this:
```js
const WrapperComponent = ( { children, className } ) => inline
	? <span className={ className }>{ children }</span>
	: <div className={ className }>{ children }</div>;
return <WrapperComponent className={ classes }> { this.renderContent() }</WrapperComponent>;
```

But as you can see, it's more convoluted and less readable than what we have right now.

In my opinion there's nothing more we can do about it.

Reply:
That's what happens when you comment with untested code 😉 Sorry about that. This should work:

```
const WrapperComponent = inline ? 'span' : 'div';

return <WrapperComponent className={ classes }>{ this.renderContent() }</WrapperComponent>;

```

Let me know what you think.

Reply:
This works! Thanks! 👍